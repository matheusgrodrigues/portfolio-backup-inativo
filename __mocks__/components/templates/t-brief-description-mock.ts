import { oHeaderAvatarMock } from "../organism/o-header-mock";

export const tBriefDescriptionMock = {
  avatarBody: oHeaderAvatarMock,
  textName: "Matheus Gomes",
  linkedinUrl: "https://www.linkedin.com/in/matheusgomes/",
  githubUrl: "https://github.com/matheusgrodrigues",
};
