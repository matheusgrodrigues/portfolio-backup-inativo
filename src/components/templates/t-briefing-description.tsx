import styled from "styled-components";

// Atoms
import { AAvatar } from "../atoms";
import { AIcon } from "../atoms/a-icon";
import { ADisplay } from "../atoms/a-display";
import { AText } from "../atoms/a-text";
import { AButton } from "../atoms/a-button";

const TBriefDescriptionStyled = styled.div``;

interface ITBriefDescription {}

export const TBriefDescription = ({}: ITBriefDescription) => {
  return (
    <TBriefDescriptionStyled>
      <AAvatar src="/images/avatar.jpeg" alt="" $variant="md" />
      <AIcon icon="sun" height={32} />

      <ADisplay $variant="lg" fontWeight="semibold" color="gradient">
        Display
      </ADisplay>

      <AText $variant="xl" fontWeight="semibold" color="gray50">
        A-Text
      </AText>

      <AButton
        $variant="lg"
        $background="color_white"
        $fontSize="fs_textMd"
        $fontWeight="fw_semibold"
        $color="color_gray500"
      >
        AButton
      </AButton>
    </TBriefDescriptionStyled>
  );
};
