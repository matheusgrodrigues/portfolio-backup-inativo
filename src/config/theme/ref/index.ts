import { colors } from './color'
import { spacing } from './spacing'
import { font } from './font'
import { bp } from './breakpoint'

export { colors, spacing, font, bp }
