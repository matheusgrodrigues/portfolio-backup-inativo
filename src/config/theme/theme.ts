const fontFamily = {
    fontFamily_inter: 'Inter',
};

export type FontSizeName = 'fontSize_textSm' | 'fontSize_textMd' | 'fontSize_textXl' | 'fontSize_displayLg';

const fontSize = {
    fontSize_textSm: '14px',
    fontSize_textMd: '16px',
    fontSize_textXl: '20px',
    fontSize_displayLg: '48px',
};

export type FontWeightName = 'fontWeight_semibold' | 'fontWeight_regular' | 'fontWeight_medium';

const fontWeight = {
    fontWeight_regular: '400',
    fontWeight_medium: '500',
    fontWeight_semibold: '600',
};

export type ColorName =
    | 'color_gradient_primary600'
    | 'color_primary500'
    | 'color_primary600'
    | 'color_gray500'
    | 'color_gray900'
    | 'color_gray50'
    | 'color_white'
    | 'color_black'
    | 'color_transparent';

const colors = {
    color_gradient_primary600: `linear-gradient(90deg, #7f56d9 0%, #9e77ed 100%);`,
    color_primary500: '#9e77ed',
    color_primary600: '#7f56d9',
    color_gray500: '#667085',
    color_gray900: '#101828',
    color_gray50: '#f9fafb',
    color_white: '#ffffff',
    color_black: '#000000',
    color_transparent: 'transparent',
};

export type SpacingName =
    | 'spacing_4'
    | 'spacing_8'
    | 'spacing_12'
    | 'spacing_16'
    | 'spacing_20'
    | 'spacing_24'
    | 'spacing_32'
    | 'spacing_40'
    | 'spacing_48'
    | 'spacing_64'
    | 'spacing_80'
    | 'spacing_96'
    | 'spacing_128'
    | 'spacing_160'
    | 'spacing_192'
    | 'spacing_224'
    | 'spacing_256';

const spacing = {
    spacing_4: '4px',
    spacing_8: '8px',
    spacing_12: '12px',
    spacing_16: '16px',
    spacing_20: '20px',
    spacing_24: '24px',
    spacing_32: '32px',
    spacing_40: '40px',
    spacing_48: '48px',
    spacing_64: '64px',
    spacing_80: '80px',
    spacing_96: '96px',
    spacing_128: '128px',
    spacing_160: '160px',
    spacing_192: '192px',
    spacing_224: '224px',
    spacing_256: '256px',
};

type BreakpointName = 'breakpoint_sm' | 'breakpoint_md' | 'breakpoint_lg' | 'breakpoint_xl';

const breakpoint = {
    breakpoint_sm: '640px',
    breakpoint_md: '768px',
    breakpoint_lg: '1024px',
    breakpoint_xl: '1280px',
};

export const bpHelper = (bp: BreakpointName, content: string) =>
    `@media screen and (min-width: ${breakpoint[bp]}) {
    ${content}
  }`;

export type ThemeName = 'light' | 'dark';

type Theme = {
    name: ThemeName;
    ref: {
        breakpoint: typeof breakpoint;
        fontFamily: typeof fontFamily;
        fontWeight: typeof fontWeight;
        fontSize: typeof fontSize;
        spacing: typeof spacing;
        colors: typeof colors;
    };
};

declare module 'styled-components' {
    export interface DefaultTheme extends Theme {}
}

export const themeLight = {
    name: 'light',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};

export const darkTheme = {
    name: 'dark',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};
