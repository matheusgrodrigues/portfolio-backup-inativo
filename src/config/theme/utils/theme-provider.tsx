'use client'

import { ThemeProvider as StyledThemeProvider } from 'styled-components'

export const ThemeProvider = StyledThemeProvider
