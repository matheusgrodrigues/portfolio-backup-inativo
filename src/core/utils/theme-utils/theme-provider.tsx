'use client';

import { ThemeProvider as StyledThemeProvider } from 'styled-components';

const ThemeProvider = StyledThemeProvider;

export default ThemeProvider;
