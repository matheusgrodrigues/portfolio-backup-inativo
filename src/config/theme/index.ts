import { colors, colorsDark, font, spacing, bp } from './ref'

export const bpHelper = (bp: string, content: string) => {
    return `
  @media screen and (min-width: ${bp}) {
    ${content}
  }
  `
}

export const themeLight = {
    ref: { colors, font, spacing, bp },
}

export const darkTheme = {
    ...themeLight,
    ref: { ...themeLight.ref, colors: { ...colorsDark } },
}
