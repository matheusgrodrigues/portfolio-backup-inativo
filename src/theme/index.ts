import { por_ref_colors, por_ref_colorsDark } from "./ref";

const theme = {
  ref: { por_ref_colors },
  sys: {},
  comp: {},
};

export const darkTheme = {
  ...theme,
  ref: { ...theme.ref, por_ref_colors: { ...por_ref_colorsDark } },
};
