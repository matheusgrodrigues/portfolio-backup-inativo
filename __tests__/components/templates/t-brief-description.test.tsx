import renderer from "react-test-renderer";

// Template
import { TBriefDescription } from "@/src/components/templates/t-briefing-description";

// Theme
import { ThemeProvider } from "@/src/theme/helpers/theme-provider";
import { themeLight } from "@/src/theme";

describe("Deve renderizar o TBriefDescription, corretamente", () => {
  it("Deve preservar a estrutura visual do TBriefDescription", () => {
    const three = renderer
      .create(
        <ThemeProvider theme={themeLight}>
          <TBriefDescription />
        </ThemeProvider>
      )
      .toJSON();

    expect(three).toMatchSnapshot();
  });
});
