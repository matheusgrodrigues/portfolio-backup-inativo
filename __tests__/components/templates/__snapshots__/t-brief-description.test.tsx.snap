// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o TBriefDescription, corretamente Deve preservar a estrutura visual do TBriefDescription 1`] = `
<div
  className="sc-eDLKEg"
>
  <h1
    className="sc-beyTiQ ezTqTU"
    color="gradient"
    data-testid="a-display"
    fontWeight="semibold"
  >
    Programador Frontend
  </h1>
  <h1
    className="sc-beyTiQ hNSGeZ"
    color="gray900"
    data-testid="a-display"
    fontWeight="semibold"
  >
    apaixonado por criação de interfaces inovadoras
  </h1>
  <p
    className="sc-guDMob eannY"
    color="gray50"
    data-testid="a-text"
    fontWeight="semibold"
  >
    Visite meu perfil no Linkedin e explore meus projetos no GitHub para descobrir como minhas habilidades podem agregar valor á sua equipe.
  </p>
  <button
    className="sc-dmyDmy fRZSRm"
    onClick={[MockFunction]}
  >
    Download CV
  </button>
  <button
    className="sc-dmyDmy fRZSRm"
    onClick={[MockFunction]}
  >
    Entre em Contato
  </button>
</div>
`;
